packages/nodecg-utility-obs/src/index.ts

```diff
@@ -158,26 +158,68 @@ export class OBSUtility extends OBSWebSocket {
 		});
 
 		nodecg.listenFor(`${namespace}:transition`, async ({name, duration, sceneName} = {}, callback) => {
-			if (sceneName) {
+			if (studioMode.value) {
+				// If in studio mode, set the preview scene, and then transition to it
+
+				if (sceneName) {
+					try {
+						await this.send('SetPreviewScene', {'scene-name': sceneName});
+					} catch (error) {
+						log.error('Error setting preview scene for transition:', error);
+						if (callback && !callback.handled) {
+							callback(error);
+						}
+						return;
+					}
+				}
+
 				try {
-					await this.send('SetPreviewScene', {'scene-name': sceneName});
+					await this._transition(name, duration);
 				} catch (error) {
-					log.error('Error setting preview scene for transition:', error);
+					log.error('Error transitioning:', error);
 					if (callback && !callback.handled) {
 						callback(error);
 					}
 					return;
 				}
-			}
+			} else {
+				// If not in studio mode, set the transition params and then set the scene
+
+				if (name) {
+					try {
+						await this.send('SetCurrentTransition', { "transition-name": name });
+					} catch (error) {
+						log.error('Error setting scene for transition:', error);
+						if (callback && !callback.handled) {
+							callback(error);
+						}
+						return;
+					}
+				}
 
-			try {
-				await this._transition(name, duration);
-			} catch (error) {
-				log.error('Error transitioning:', error);
-				if (callback && !callback.handled) {
-					callback(error);
+				if (duration) {
+					try {
+						await this.send('SetTransitionDuration', { duration: duration });
+					} catch (error) {
+						log.error('Error setting duration for transition:', error);
+						if (callback && !callback.handled) {
+							callback(error);
+						}
+						return;
+					}
+				}
+
+				try {
+					this.replicants.transitioning.value = true;
```

Comment:
Can you add a comment explaining how `transitioning` gets set back to `false`? I'm assuming it's in the `SwitchScenes` event handler, but it's good to be explicit.

Reply:
I don't mind doing that, but the existing instance (_transition() method) doesn't seem to state how its setting set back to false, unless I'm missing something. Regardless notifying of potential side effects is a good call. Yeah it is the SwitchScenes event.